processor/intervalprocessor/processor.go

```diff
@@ -37,47 +45,52 @@ type Processor struct {
 	histogramLookup    map[identity.Stream]pmetric.HistogramDataPoint
 	expHistogramLookup map[identity.Stream]pmetric.ExponentialHistogramDataPoint
 	summaryLookup      map[identity.Stream]pmetric.SummaryDataPoint
-
-	config *Config
-
-	nextConsumer consumer.Metrics
 }
 
 func newProcessor(config *Config, log *zap.Logger, nextConsumer consumer.Metrics) *Processor {
 	ctx, cancel := context.WithCancel(context.Background())
 
-	return &Processor{
-		ctx:    ctx,
-		cancel: cancel,
-		logger: log,
-
-		stateLock: sync.Mutex{},
-
-		md:                 pmetric.NewMetrics(),
-		rmLookup:           map[identity.Resource]pmetric.ResourceMetrics{},
-		smLookup:           map[identity.Scope]pmetric.ScopeMetrics{},
-		mLookup:            map[identity.Metric]pmetric.Metric{},
-		numberLookup:       map[identity.Stream]pmetric.NumberDataPoint{},
-		histogramLookup:    map[identity.Stream]pmetric.HistogramDataPoint{},
-		expHistogramLookup: map[identity.Stream]pmetric.ExponentialHistogramDataPoint{},
-		summaryLookup:      map[identity.Stream]pmetric.SummaryDataPoint{},
+	numPartitions := int(config.Interval.Seconds())
```

Comment:
Oh, I think this should be either hardcoded to something like 60. Or a configurable. If someone chooses a large interval like 15 minutes, we don't want to create 900 partitions as a result

Reply:
This might sound like a dumb question, but why 900 partitions is a problem? 😅

Reply:
I think it's just overkill. Also, I don't think we want to export every second. It's a lot of work to be doing every second.

I guess this highlights the issue with hardcoding or such. If the interval is short, like 15 seconds, we probably don't need to partition at all. But if it's 15 minutes or an hour, then partitioning will be very impactful. But again, we don't want to export every second. 

IE, maybe we calculate the number of intervals to have, based on some "ideal" export frequency. (no idea what value to choose for that).

For example, say we choose the "ideal" export interval to be 15 seconds. If the "processor interval" is less than 15 seconds, we only have one partition. Between 15 and 30 seconds, two intervals. Between 30 and 45 seconds, three intervals. Etc.

Reply:
> I think it's just overkill. Also, I don't think we want to export every second. It's a lot of work to be doing every second.

Ok I see your point. I guess our approach here is switching where the heavy computation load is happening.

* No partitions = heavy load on who is receiving the aggregated metrics since everything arrives at the same time
* More partitions = Smoother load on the next consumer, but higher load on intervalprocessor

> IE, maybe we calculate the number of intervals to have, based on some "ideal" export frequency. (no idea what value to choose for that).

Not sure if we'll be able to find a one-fits-all here, it all depends on the characteristics of the incoming metrics 😕.